lib/puppet/provider/cisco_vpc_domain/cisco.rb

```diff
@@ -51,10 +51,15 @@
     :system_mac,
     :system_priority,
   ]
+  # auto_recovery and peer_gateway need to be set before non_boolean
+  # properties like auto_recovery_reload_delay and peer_gateway_exclude_vlan
+  # so making it into a new array
   VPC_BOOL_PROPS1 = [
     :auto_recovery,
     :peer_gateway,
   ]
+  # these properties like port_channel_limit need to be set after non_boolean
+  # properties like fabricpath_emulated_switch_id
```

Comment:
I think a single comment is all you need here. How about this instead:

```
  # There are multiple BOOL arrays due to dependency requirements. The order of
  # these property arrays is important for proper processing by properties_set.
  VPC_BOOL_PROPS1 = [
    :auto_recovery,
    :peer_gateway,
  ]
   VPC_BOOL_PROPS2 = [
      :fabricpath_multicast_load_balance,
      :graceful_consistency_check,
     :layer3_peer_routing,
     :port_channel_limit,
      :self_isolation,
      :shutdown,
    ]
  VPC_PROPS = VPC_BOOL_PROPS1 + VPC_NON_BOOL_PROPS + VPC_BOOL_PROPS2
```

That's all I was looking for. :-)